### PR TITLE
Rebuild just the card instead of entire view

### LINKED ISSUE #2443: Lovelace flash on load (0.84 + 0.85)
In 0.84 lovelace load would initially show the non lovelace UI and then redirect to lovelace.
In 0.85 this somehow changed (or maybe it didn't) but now it loads and immediately refreshes.
Note: happens in both RTL & LTR so language is irrelevant.

Video:
https://drive.google.com/file/d/1KnU2qygIyNRCThpcgJ1NgY0LP0Ui8oW_/view?usp=sharing

### PR DESCRIPTION
**Pull request by bramkragten**
For custom cards, prevent flickering on loading.

Fixes https://github.com/home-assistant/home-assistant-polymer/issues/2443

### CODE REVIEW

**Review by balloob — APPROVED** [review]

**Comment by balloob** [comment]
Made some tweaks, what do you think @bramkragten ?

Realized that composed path is difficult because we need to find the right config again. Ended up with adding just listeners to the cards to update themselves.

**Comment by bramkragten** [comment]
Looks good, a combination of both would be ideal, only rebuild the card that is loaded and not also it's parent and all children. We could add this same logic to the stack and conditional card and handle it there? Or would that be overkill?

File: src/panels/lovelace/hui-root.ts [hunk 1]
```diff
@@ -298,7 +291,7 @@ class HUIRoot extends hassLocalizeLitMixin(LitElement) {
       </app-header>
       <div id='view' class="${classMap({
         "tabs-hidden": this.lovelace!.config.views.length < 2,
-      })}" @rebuild-view='${this._debouncedConfigChanged}'></div>
+      })}" @ll-rebuild='${this._debouncedConfigChanged}'></div>
```
Comment (thomasloven) [inline src/panels/lovelace/hui-root.ts]: Not sure how events work...
Will this get triggered every time the ll-rebuild event is fired, or is it caught and eaten by the handler in hui-view?

Comment (bramkragten) [reply src/panels/lovelace/hui-root.ts]: If a card takes it, it will stop it from reaching the view. Only in edit mode we let it reach the view

Comment (thomasloven) [reply src/panels/lovelace/hui-root.ts]: Good

Comment (balloob) [reply src/panels/lovelace/hui-root.ts]: It's done by calling `event.stopPropagation()` in the event handler for the card.

https://developer.mozilla.org/en-US/docs/Web/API/Event/stopPropagation

**Comment by balloob** [comment]
Conditional cards it won't matter, it is very lightweight. For stacks it could be beneficial.

btw one downside of this approach is that by not rebuilding the view, the size is different, and so cards are not positioned correctly.

**Comment by daniele-athome** [comment]
> btw one downside of this approach is that by not rebuilding the view, the size is different, and so cards are not positioned correctly.

I believe I'm having this issue: some cards fill their content dinamically (i.e. after being laid out) and change height. Depending on the random loading order (like whichever finishes first), I get a 2-columns layout instead of 3-columns like it should (my YAML has 3 vertical stacks in the view).
Is there a way to fix this using configuration?

By the way, here is my config: https://pastebin.com/84mx95Nw

**Comment by daniele-athome** [comment]
Never mind, I switched to layout card :-)

**Pull request merged**